File: 1247_minimum_swaps_to_make_strings_equal/solution.go
// LeetCode 1247 - Minimum Swaps to Make Strings Equal
// https://leetcode.com/problems/minimum-swaps-to-make-strings-equal/

func minimumSwap(s1 string, s2 string) int {
	xy, yx := 0, 0
	for i := 0; i < len(s1); i++ {
		if s1[i] == 'x' && s2[i] == 'y' {
			xy++
		} else if s1[i] == 'y' && s2[i] == 'x' {
			yx++
		}
	}
	if (xy+yx)%2 != 0 {
		return -1
	}
	return xy/2 + yx/2 + 2*(xy%2)
}

File: 1247_minimum_swaps_to_make_strings_equal/Solution.swift
// LeetCode 1247 - Minimum Swaps to Make Strings Equal
// https://leetcode.com/problems/minimum-swaps-to-make-strings-equal/

class Solution {
    func minimumSwap(_ s1: String, _ s2: String) -> Int {
        let a = Array(s1), b = Array(s2)
        var xy = 0, yx = 0
        for i in 0..<a.count {
            if a[i] == "x" && b[i] == "y" { xy += 1 }
            if a[i] == "y" && b[i] == "x" { yx += 1 }
        }
        if (xy + yx) % 2 != 0 { return -1 }
        return xy / 2 + yx / 2 + (xy % 2) * 2
    }
}
